x-pack/plugins/alerts/server/alert_type_registry.ts

```diff
@@ -144,21 +169,52 @@ function normalizedActionVariables(actionVariables: AlertType['actionVariables']
   };
 }
 
-function validateActionGroups(alertTypeId: string, actionGroups: ActionGroup[]) {
-  const reservedActionGroups = intersection(
-    actionGroups.map((item) => item.id),
-    getBuiltinActionGroups().map((item) => item.id)
+function augmentActionGroupsWithReserved<
+  Params extends AlertTypeParams,
+  State extends AlertTypeState,
+  InstanceState extends AlertInstanceState,
+  InstanceContext extends AlertInstanceContext
+>(
+  alertType: AlertType<Params, State, InstanceState, InstanceContext>
+): NormalizedAlertType<Params, State, InstanceState, InstanceContext> {
+  const reservedActionGroups = getBuiltinActionGroups(alertType.recoveryActionGroup);
+  const { id, actionGroups, recoveryActionGroup } = alertType;
+
+  const activeActionGroups = new Set(actionGroups.map((item) => item.id));
+  const intersectingReservedActionGroups = intersection(
+    [...activeActionGroups.values()],
+    reservedActionGroups.map((item) => item.id)
   );
-  if (reservedActionGroups.length > 0) {
+  if (recoveryActionGroup && activeActionGroups.has(recoveryActionGroup.id)) {
+    throw new Error(
+      i18n.translate(
+        'xpack.alerts.alertTypeRegistry.register.customRecoveryActionGroupUsageError',
+        {
+          defaultMessage:
+            'Alert type [id="{id}"] cannot be registered. Action group [{actionGroup}] cannot be used as both a recovery and an active action group.',
+          values: {
+            actionGroup: recoveryActionGroup.id,
+            id,
+          },
+        }
+      )
+    );
+  } else if (intersectingReservedActionGroups.length > 0) {
     throw new Error(
-      i18n.translate('xpack.alerts.alertTypeRegistry.register.reservedActionGroupUsageError', {
+      i18n.translate('xpack.alerts.alertTypeRegistry.register.recoveryActionGroupUsageError', {
```

Comment:
Should this stay `reservedActionGroupUsageError`

Reply:
Oh, dammit, yet!

Reply:
My brain read it as resolved